app.rb

```diff
@@ -32,25 +32,63 @@ def list_all_people
     end
   end
 
-  def create_student(age, classroom, name, permission)
+  def create_person
+    puts('Do you want to create a student (1) or a teacher (2)? [Input the number]: ')
+    person_type = gets.chomp.to_i
+    case person_type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts('person option not valid')
+    end
+  end
+
+  def create_student
+    puts('Enter age: ')
+    age = gets.chomp
+    puts('Enter name: ')
+    name = gets.chomp
+    puts('Enter classroom: ')
+    classroom = gets.chomp
+    puts('Has parent permission? [Y/N]: ')
+    response = gets.chomp
+    permission = response.downcase == 'y'
     student = Student.new(age, classroom, name, parent_permission: permission)
     puts("Created student id: #{student._id} name: #{student._name}")
     @people << student
   end
 
-  def create_teacher(age, specialization, name)
+  def create_teacher
+    puts('Enter age: ')
+    age = gets.chomp
+    puts('Enter specialization: ')
+    specialization = gets.chomp
+    puts('Enter name: ')
+    name = gets.chomp
     teacher = Teacher.new(age, specialization, name)
     @people << teacher
     puts("Created teacher id: #{teacher._id} name: #{teacher._name}")
   end
```

Comment:
- [ ] In `create_teacher` method you are getting input from users and creating the teacher, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.